crates/cheatcodes/src/evm.rs

```diff
@@ -89,7 +90,24 @@ impl Cheatcode for loadCall {
         let Self { target, slot } = *self;
         ensure_not_precompile!(&target, ccx);
         ccx.ecx.load_account(target)?;
-        let val = ccx.ecx.sload(target, slot.into())?;
+        let mut val = ccx.ecx.sload(target, slot.into())?;
+
+        if val.is_cold && val.data.is_zero() {
+            let rand_value = ccx.state.rng().gen();
```

Comment:
should not gen unless either is true

Reply:
yep, changed in https://github.com/foundry-rs/foundry/pull/8807/commits/7e69723609d13ceb3ce744c0f7bbdd3e156f1296